server/src/data_sources/redis_provider.rs

```diff
@@ -38,8 +40,12 @@ impl RedisProvider {
 impl EdgeSource for RedisProvider {}
 impl EdgeSink for RedisProvider {}
 
-fn build_features_key(token: &String) -> String {
-    format!("{FEATURE_PREFIX}{token}")
+fn build_features_key(token: &EdgeToken) -> String {
+    token
+        .environment
+        .as_ref()
+        .map(|environment| format!("{FEATURE_PREFIX}{environment}"))
+        .expect("Tying to resolve features for a token that hasn't been validated")
```

Comment:
```suggestion
        .expect("Resolving features for a validated token")
```
Just words, but when I read the `expect` (might be because I'm unfamiliar with Rust yet), it seems that you're expecting a non validated token...

Reply:
Yeah, that's fair. Expect should loosely read as "I expected this value and if I didn't get it, crash with the enclosed error message"

Reply:
I also agree that `expect` is a bit confusing readability wise. It's almost like it should be a `except` instead.